upup/pkg/fi/cloudup/openstacktasks/floatingip.go

```diff
@@ -70,6 +71,36 @@ func findL3Floating(cloud openstack.OpenstackCloud, opts l3floatingip.ListOpts)
 	return result, nil
 }
 
+func (e *FloatingIP) findServerFloatingIP(context *fi.Context, cloud openstack.OpenstackCloud) (*string, error) {
+	var result *string
+	_, err := vfs.RetryWithBackoff(readBackoff, func() (bool, error) {
```

Comment:
This logic might move underneath the cloud object and be extracted from `GetApiIngressStatus` as well, into a common call.